src/core/or/circuitpadding.c

```diff
@@ -502,8 +503,8 @@ circpad_machine_sample_delay(circpad_machine_state_t *mi)
     return bin_start;
```

Comment:
It's also a bug if bin_start == bin_end, because crypto_fast_rng_uint64_range() will assert and crash.